pkg/cli/debug/debug.go

```diff
@@ -962,6 +957,40 @@ func (o *DebugOptions) approximatePodTemplateForObject(object runtime.Object) (*
 	return nil, fmt.Errorf("unable to extract pod template from type %v", reflect.TypeOf(object))
 }
 
+func (o *DebugOptions) logsOptions(pod *corev1.Pod) error {
+	return logs.LogsOptions{
+		Object: pod,
+		Options: &corev1.PodLogOptions{
+			Container: o.Attach.ContainerName,
+			Follow:    true,
+		},
+		RESTClientGetter: o.RESTClientGetter,
+		ConsumeRequestFn: logs.DefaultConsumeRequest,
+		IOStreams:        o.IOStreams,
+		LogsForObject:    o.LogsForObject,
+	}.RunLogs()
+}
+
```

Comment:
No change in the above, I just moved it to a function so as not to repeat
Below, I've added the `getPodLogsForError` to return the podlogs as error in case of non-zero container exit.